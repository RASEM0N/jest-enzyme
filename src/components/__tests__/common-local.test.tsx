import 'jsdom-global/register'
import React from 'react'
import CommonLocale from '../CommonLocal'
import { shallow, render, mount } from 'enzyme'

describe('CommonLocal.tsx', () => {
    it('shapshot - render', () => {
        const component = mount(<CommonLocale />)

        expect(component).toMatchSnapshot()
    })

    it('click on the increment button', () => {
        const component = mount(<CommonLocale />)

        const h2 = component.find('h2')
        const btnIncr = component.find('#incr')
        // const btnDescr = component.find('#decr')

        btnIncr.simulate('click')

        expect(h2.text()).toBe(2)
    })
})
