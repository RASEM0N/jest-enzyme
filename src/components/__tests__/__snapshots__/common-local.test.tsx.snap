// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonLocal.tsx shapshot - render 1`] = `
"<Common>
  <div>
    <h2>
      0
    </h2>
    <button id=\\"incr\\" onClick={[Function: incr]}>
      +1
    </button>
    <button id=\\"decr\\" onClick={[Function: decr]}>
      -1
    </button>
  </div>
</Common>"
`;
